server/anomaly_scanner.go

```diff
@@ -58,23 +58,23 @@ func (s *AnomalyScanner) Run(ctx context.Context, wg *sync.WaitGroup) {
 
 				ctx := context.Background()
 
-				environmentFind := &api.EnvironmentFind{}
-				environmentList, err := s.server.EnvironmentService.FindEnvironmentList(ctx, environmentFind)
+				envFind := &api.EnvironmentFind{}
+				envRawList, err := s.server.EnvironmentService.FindEnvironmentList(ctx, envFind)
 				if err != nil {
 					s.l.Error("Failed to retrieve instance list", zap.Error(err))
 					return
 				}
 
 				backupPlanPolicyMap := make(map[int]*api.BackupPlanPolicy)
-				for _, env := range environmentList {
-					policy, err := s.server.PolicyService.GetBackupPlanPolicy(ctx, env.ID)
+				for _, envRaw := range envRawList {
```

Comment:
Shall we compose envRawList to envList here first? Then the following code doesn't need a change.

Reply:
But this will eagerly compose all the environments, which is worse than the current lazy compose behavior.

Reply:
The number of environments are limited :-) are we talking about optimizing O(~5)?

Reply:
Well, the specific problem here is not about performance.
The original code here does not use any relation (pointer) fields of `Environment`, so the composition is useless.

Reply:
Fixed, PTAL.